packages/nimble-components/src/tree-item/index.ts

```diff
@@ -25,60 +25,73 @@ declare global {
  * Generates HTML Element: \<nimble-tree-item\>
  *
  */
-export class TreeItem extends FoundationTreeItem {
-    private treeView: TreeView | null = null;
+export class TreeItem extends FoundationTreeItem implements ISelectableSubtree {
+    /**
+     * @internal
+     */
+    @observable
+    public subtreeHasSelection = false;
+
+    private childNotifiers: Notifier[] = [];
 
     public override connectedCallback(): void {
         super.connectedCallback();
-        this.addEventListener('selected-change', this.handleSelectedChange);
-        this.treeView = this.getParentTreeView();
-        if (this.treeView && this.selected) {
-            this.setGroupSelectionOnRootParentTreeItem(this);
-        }
+        this.observeChildren();
+        this.addEventListener('selected-change', () => this.onSelectedChange());
     }
 
-    public override disconnectedCallback(): void {
-        super.disconnectedCallback();
-        this.removeEventListener('selected-change', this.handleSelectedChange);
-        this.treeView = null;
+    /**
+     * @internal
+     */
+    public handleChange(_source: unknown, _args: unknown): void {
+        this.updateSubtreeHasSelection();
     }
 
-    // This prevents the toggling of selected state when a TreeItem is clicked multiple times,
-    // which is what the FAST TreeItem allows
```

Comment:
I don't know why this comment says this. This handler doesn't do that. I think the TreeView's click handler is responsible for that behavior.